web/packages/teleport/src/Roles/RoleEditor/StandardEditor/RequiresResetToStandard.tsx

```diff
@@ -16,14 +16,151 @@
+ * the standard editor. Provides a dialog that explains the details of what's
+ * not supported and lets the user commit the proposed changes.
+ */
+export const RequiresResetToStandard = ({
+  conversionErrors,
+  onReset,
+}: {
+  conversionErrors: ConversionErrorGroup[];
+  /**
+   * Called if the user decides to reset the role to be compatible with the
+   * standard model.
+   */
+  onReset(): void;
+}) => {
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const handleReviewClick = () => {
+    setDialogOpen(true);
+  };
+  const handleCancelClick = () => {
+    setDialogOpen(false);
+  };
+  return (
+    <Info>
+      <Text>
+        This role is too complex to be edited in the standard editor. To
+        continue editing, go back to YAML editor or reset the affected fields to
+        standard settings.
+      </Text>
+      <ButtonSecondary size="large" my={2} onClick={handleReviewClick}>
+        Review and Reset to Standard Settings
+      </ButtonSecondary>
+      <Dialog open={dialogOpen}>
+        <DialogHeader mb={4}>
+          <H2>Review and Reset to Standard Settings</H2>
+        </DialogHeader>
+        <DialogContent mb={3}>
+          <P2>
+            To support editing this role in the standard editor, the following
+            fields will be altered. Please review the list carefully before
+            continuing.
+          </P2>
+
+          {conversionErrors.map(group => (
+            // group.type is unique per contract of
+            // `groupAndSortConversionErrors()`.
+            <ErrorGroupSection key={group.type} group={group} />
+          ))}
+        </DialogContent>
+        <Flex gap={3}>
+          <ButtonPrimary block size="large" onClick={onReset}>
+            Reset to Standard Settings
+          </ButtonPrimary>
+          <ButtonSecondary block size="large" onClick={handleCancelClick}>
+            Cancel
+          </ButtonSecondary>
+        </Flex>
+      </Dialog>
+    </Info>
+  );
+};
+
+const ErrorGroupSection = ({
+  group: { type, errors },
+}: {
+  group: ConversionErrorGroup;
+}) => {
+  return (
+    <>
+      <ErrorGroupSectionHeader type={type} />
+      <ul>
+        {errors.map((error, i) => (
+          <li key={`${i}:${error.path}`}>
+            <ErrorMessage error={error} />
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+};
+
+const ErrorGroupSectionHeader = ({ type }: { type: ConversionErrorType }) => {
+  switch (type) {
+    case ConversionErrorType.UnsupportedField:
+      return <P2>The following fields are unsupported and will be deleted:</P2>;
+    case ConversionErrorType.UnsupportedValue:
+      return (
+        <P2>
+          The following fields have unsupported values and will be deleted:
+        </P2>
+      );
+    case ConversionErrorType.UnsupportedValueWithReplacement:
+      return (
+        <P2>
+          The following fields have unsupported values and will be replaced:
```

Comment:
Is the "unsupported values" something we can identify and report to user so they can update their input?

Reply:
(Oh, never mind, I think we'd only need these for UnsupportedValue and UnsupportedValueWithReplacement. That being said, it complicates my model a bit.)

Reply:
Error path is informative but not actionable. If we can tell about both the unsupported value and the reason it is not supported, that will help user to correct the input value. 

Alternatively, we can also provide a links to docs on that particular role spec so that user would learn supported values.

Reply:
> That being said, it complicates my model a bit

Yeah let's not worry if this begs more changes for now. We can revisit later when the time permits.

Reply:
Additional reflection, to note before it perishes from my mind: investing more time in polishing this feature will be less impactful IMO than investing more time in supporting things that we don't support. :)